corehq/apps/accounting/invoicing.py

```diff
@@ -427,6 +427,10 @@ def _email_invoice(self):
 
 
 def should_create_invoice(subscription, domain, invoice_start, invoice_end):
+    from corehq.apps.domain.models import Domain
+    if Domain.get_by_name(domain) is None:
```

Comment:
I think you could make use of [`deleted_domain_exists`](https://github.com/dimagi/commcare-hq/blob/8fb14807fbe323b70fef56d3005588d46954b0d6/corehq/apps/domain/dbaccessors.py#L126-L128) and change this to maybe

```python
if not domain_exists(domain) and deleted_domain_exists(domain):
```

or something like that to just have the condition be super explicit and not vulnerable to any unknown edge cases